spark3/src/test/java/org/apache/iceberg/actions/TestRemoveOrphanFilesAction3.java

```diff
@@ -134,6 +135,24 @@ public void testSparkSessionCatalogHadoopTable() throws Exception {
         results.contains("file:" + location + "/data/trashfile"));
   }
 
+  @Test
+  public void testSparkSessionCatalogHiveWrongConfig() {
+    spark.conf().set("spark.sql.catalog.spark_catalog", "org.apache.iceberg.spark.SparkSessionCatalog");
+    spark.conf().set("spark.sql.catalog.spark_catalog.type", "hive");
+    spark.conf().set("spark.sql.catalog.spark_catalog.uri", "thrift://localhost:9083");
+    if (spark.version().equalsIgnoreCase("3.0.3")) {
```

Comment:
I think that this can just check for `spark.version().startsWith("3.0"))`.

If I'm not mistaken, I think that all of Spark 3.0.x (and not just 3.0.3) will have an issue.

Reply:
Not sure I understand the Class Cast here,
Also we have some Spark3VersionUtil classes you may want to use